torch/lib/c10d/comm.cpp

```diff
@@ -87,4 +86,17 @@ void broadcast_coalesced(
   }
 }
 
+std::vector<at::Tensor> GradBucket::getPerParameterTensors() const {
+  std::vector<at::Tensor> per_parameter_tensors;
+  size_t num_parameters = offsets_.size();
```

Comment:
It looks like in constructor, offsets, lengths, and sizes_vec are optional arguments and could possibly be empty. It looks like only the reducer initializes the GradBucket and passes in these values, but I'm wondering if we should throw an error here if those arguments were not specified. For example I think that this would return just an empty vector if `offsets_` wasn't provided, which is probably not the result we want to return.

Reply:
Good point! Let me address this in a follow-up PR, as it's a separate topic. I plan to remove these default values and polish the constructor API.

Previously I set some default values probably because it can minimize the changes in some references (probably some tests).